File: Web/app/objects/productList/productItem.ts
import { Product } from "../product/product";
/**
 * Created by vadim.m on 11/16/2017.
 */
export class ProductItem {
    public Id: string;
    public ProductListId: string;
    public ProductId: number;
    public VariantId: number;
    public Amount: number;
    public Product:Product;

    public constructor(init?:Partial<ProductItem>) {
        Object.assign(this, init);
    }
}

File: BL.Objects/ProductItem/ProductItem.cs
using System;

namespace BL.Objects.ProductItem
{
    public class ProductItem
    {
        public Guid? Id { get; set; }

        public Guid ProductListId { get; set; } 

        public long ProductId { get; set; }

        public long VariantId { get; set; }

        public int Amount { get; set; }

        public Product.Product Product { get; set; }
    }
}
